UnityProject/Assets/Scripts/Objects/Research/Artifacts/Effects/AreaEffects/AreaEffectBase.cs

```diff
@@ -0,0 +1,99 @@
+using System.Linq;
+using UnityEngine;
+using System.Collections.Generic;
+using HealthV2;
+
+namespace Systems.Research
+{
+	public class AreaEffectBase : ArtifactEffect
+	{
+		public EffectShapeType effectShapeType = EffectShapeType.Square;
+
+		public int coolDown = 10;
+
+		[Tooltip("How far this artifact will effect players from")]
+		public int AuraRadius = 10;
+
+		[Tooltip("Allows the artifact to effect the head and body differently if they are wearing only partially protective gear")]
+		public bool AllowBodySplitting = false;
+
+		public virtual void DoEffectAura(GameObject centeredAround)
+		{
+			var objCenter = centeredAround.AssumedWorldPosServer().RoundToInt();
+			var shape = EffectShape.CreateEffectShape(effectShapeType, objCenter, AuraRadius);
+
+			foreach (var pos in shape)
```

Comment:
This looks expensive. Maybe Bod has another idea to find players inside the aoe

Reply:
In the past been stuff about Physics2D.Overlap also being fairly expensive, not sure if itll be more or less than this current method however.

Reply:
It was usually code on UnityStation's side that caused the issues, not overlap itself. MatrixManager.GetAt isn't exactly all that efficient and also creates a new list if for every tile that has objects, even if nothing matches the type.

Reply:
> In the past been stuff about Physics2D.Overlap also being fairly expensive, not sure if itll be more or less than this current method however.

Physics2D.Overlap will be a lot faster than checking the tiles. You'd only need to do that if it was items you were looking for as they don't have a collider so can't be found by the physics system. Players are also on there own physics layer so it's quite quick to do the overlap check.